src/atom-environment.coffee

```diff
@@ -229,6 +229,10 @@ class AtomEnvironment extends Model
 
     checkPortableHomeWritable()
 
+    # Patch the `process.env` on startup to fix the problem first documented
+    # in #4126
+    process.env = @project.getEnv()
```

Comment:
> then maybe `@project.getEnv()` could be renamed to be `@project.env`

That would lose us the current feature that packages can't muck up the environment for other packages (at least through the public API). Right now, everyone gets a copy and then they can do what they want with their copy.

I'll definitely move the first initialization further up though.